python-lib/wordcloud_generator.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+"""Module with a class to generate wordclouds based on cleaned text"""
+
+import dataiku
+from dataiku.core import dkuio
+import logging
+import matplotlib
+import matplotlib.pyplot as plt
+
+matplotlib.use("agg")
+
+import traceback
+from typing import List, AnyStr
+import pandas as pd
+import spacy.lang
+import os
+from wordcloud import WordCloud
+from collections import Counter
+from time import time
+from spacy_tokenizer import MultilingualTokenizer
+from plugin_io_utils import count_records
+import random
+
+
+class WordcloudGenerator:
+    """Class to generate multilingual wordclouds based on text data and save them as png images
+    Attributes:
+        df (pandas.DataFrame): Dataframe containing text data
+        tokenizer (MultilingualTokenizer): Tokenizer used for text processing
+        text_column (str): Name of the df column on which to compute wordclouds
+        language (str, optional): Language in which to tokenize data (for monolingual wordcloud), defaults to english.
+            Use "Detected language column" for multilingual tokenization
+        language_column (str, optional): Name of the language column
+        subchart_column (str, optional): Name of the subcharts column to compute wordclouds on, defaults to None
+        max_words (int, optional): Maximum number of words to display in wordcloud, defaults to 100
+    """
+
+    DEFAULT_MAX_WORDS = 100
+    DEFAULT_COLOR_LIST = [
+        "hsl(205,71%,41%)",
+        "hsl(214,56%,80%)",
+        "hsl(28,100%,53%)",
+        "hsl(30,100%,74%)",
+        "hsl(120,57%,40%)",
+        "hsl(110,57%,71%)",
+    ]
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        tokenizer: MultilingualTokenizer,
+        text_column: AnyStr,
+        output_folder: dataiku.core.managed_folder.Folder,
+        language: AnyStr = "en",
+        language_column: AnyStr = None,
+        subchart_column: AnyStr = None,
+        max_words: int = DEFAULT_MAX_WORDS,
+        color_list: list = DEFAULT_COLOR_LIST,
+    ):
+        # Initialization method for the MultilingualTokenizer class, with optional arguments etailed above
+
+        self.df = df
+        self.tokenizer = tokenizer
+        self.text_column = text_column
+        self.language = language
+        self.language_column = language_column
+        self.subchart_column = subchart_column
+        self.output_folder = output_folder
+        self.max_words = max_words
+        self.color_list = color_list
+
+    def _color_func(self, word, font_size, position, orientation, random_state=None, **kwargs):
+        # Return the color function used in the wordcloud
+        return random.choice(self.color_list)
+
+    def _get_wordcloud(self, frequencies, scale=6.8):
```

Comment:
scale should be a class constant.
Doing this will allow us to clearly see which parameter we could later add to the UI